var/spack/repos/builtin/packages/py-notebook/package.py

```diff
@@ -47,3 +47,8 @@ class PyNotebook(PythonPackage):
     depends_on('py-jupyter-client@5.3.1:', type=('build', 'run'), when='@6:')
     depends_on('py-terminado@0.8.1:', type=('build', 'run'), when='@6:')
     depends_on('py-ipaddress', type=('build', 'run'), when='@6: ^python@:2.8')
+
+    @when('@5.7.8')
+    def install_args(self, spec, prefix):
+        args = ['--prefix={0}'.format(prefix), '--root=/']
+        return args
```

Comment:
This version is not in Spack, and these args are already added automatically

Reply:
With the default arguments, the install fails:
```
==> Executing phase: 'install'
==> [2019-12-11-11:49:11.251112] '/mnt/build/jenkins/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/python-2.7.16-thcifv7hiw2y5pgh5isqo4oodmunmvak/bin/python2.7' '-s' 'setup.py' '--no-user-cfg' 'install' '--prefix=/mnt/build/jenkins/spack/opt/spack/linux-centos7-skylake/gcc-8.3.0/py-notebook-5.7.8-l4usuot3ns4f2mwul47zsli35aku2d6a'
running install
running bdist_egg
Aborting implicit building of eggs. Use `pip install .` to install from source.
```
Alternatively, adding a dependency on `setuptools` fixes the error.